doc/voq/bgp_voq_chassis.md

```diff
@@ -0,0 +1,164 @@
+
+Note that if more equal-cost paths are learned for a given route than the
+maximum ECMP group size, each FSI may still choose a different subset of paths
+to program.
+
+## 2.3 FRR Changes
+
+The second configuration mentioned above, “bgp bestpath peer-type
+multipath-relax”, does not currently exist in FRR and will be added as part of
+this proposal.  This is named to mirror the “bgp bestpath as-path
+multipath-relax” configuration.
+
+Configuring this will allow paths from different peer types to form an ECMP
+group.  The best path algorithm is not changed otherwise, such that if a route
+has both eBGP and iBGP paths in the ECMP group, an eBGP path will still be the
+best path.  This route will be the one advertised to eBGP peers, as would be
+the case without enabling the new configuration.  While the VoQ chassis system
+requires the eBGP and iBGP maximum-paths size to be set the same, if they were
+not, the eBGP size would be used for a mixed-type ECMP group, because the best
+path is guaranteed to be eBGP.
+
+## 2.4 CONFIG_DB Changes
+
+CONFIG_DB currently contains three tables specifying BGP neighbors:
+BGP_NEIGHBOR for typical router peers (R1 etc. in the diagram), BGP_MONITORS
+for route monitor(s), and BGP_PEER_RANGE for dynamic peers.  A fourth table
+will be added, BGP_VOQ_CHASSIS_NEIGHBOR, to include the FSI neighbors in the
+iBGP mesh.  Entries in this table will use the same schema as BGP_NEIGHBOR.
+
+This new table will allow the most natural use of the FRR configuration system,
+where the database table dictates which templates are used to generate the FRR
+configuration.
+
+## 2.5 FRR Configuration Templates
+
+BGP neighbors included in CONFIG_DB are translated from Jinja templates into
+FRR configuration by a script running in the bgp docker called “bgpcfgd”.
+
+Using the `general` templates used for BGP_NEIGHBOR peers would require adding
+new peer groups to the general templates and a number of if-statements based on
+some attribute of the BGP_NEIGHBOR entry.
+
+Instead, entries in the BGP_VOQ_CHASSIS_NEIGHBOR table will use a new set of
+`voq_chassis` templates rather than the `general` ones.  These templates will
+include new peer-groups to encapsulate the neighbor configuration, as well as
+the necessary instance-wide configuration (configuration changes 2-4 described
+earlier).
+
+## 2.6 Mingraph Changes
+
+The script that converts the minigraph into CONFIG_DB entries will be modified
+to place the VoQ chassis peers in the new BGP_VOQ_CHASSIS_NEIGHBOR table.  The
+minigraph must be changed to indicate which peers should be in the new table
+instead of the BGP_NEIGHBOR table.
+
+We propose a new optional element in the BGPRouterDeclaration that indicates
+whether this is a VoQ chassis neighbor.
+```
+      <a:BGPRouterDeclaration>
+        <a:ASN>64542</a:ASN>
+        <a:Hostname>OCPSCH0104001MS</a:Hostname>
+        <a:RouteMaps/>
+        <a:VoQChassisPeer>1</a:VoQChassisPeer>
+      </a:BGPRouterDeclaration>
+```
+
```

Comment:
Add a Cli section
1. "show ip bgp" shows both internal and external today, should we update the command to show only the external sessions and add a new command to include the internal. 
2. A flag that brings up the instance with bgp sessions shut down must not do it for the internal BGP sessions.
3. 'shutdown' should bring down only the external sessions, not the internal BGP sessions. We need an additional token to also shutdown the internal sessions if needed.